lib/src/components/NxTag/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import {ReactNode, ValidationMap, HTMLAttributes} from 'react';
+import * as PropTypes from 'prop-types';
+
+export const NX_TAG_COLORS =
+    ['light-blue', 'purple', 'pink', 'blue', 'red', 'green', 'orange', 'yellow', 'lime', 'indigo'] as const;
+export type NX_TAG_COLORS_TYPE = (typeof NX_TAG_COLORS)[number]; // See https://stackoverflow.com/a/45486495
+
+export interface Props extends Omit<HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+  children: ReactNode;
+  tagColor?: NX_TAG_COLORS_TYPE;
+  // For internal use only, this prop is for our select/deselct tag icons
+  selectedIcons?: ReactNode | null;
```

Comment:
Minor: this can be constrained more since it has to be an icon and not text or anything else
```suggestion
  selectedIcons?: ReactElement | null;
```